atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java

```diff
@@ -44,6 +44,38 @@
 
     @Value.Default
     public int poolSize() {
+        return 30;
+    }
+
+    /**
+     * The cap at which the connection pool is able to grow over the {@link #poolSize()}
+     * given high request load. When load is depressed, the pool will shrink back to its
+     * idle {@link #poolSize()} value.
+     */
+    @Value.Default
+    public int maxConnectionBurstSize() {
+        return 100;
+    }
+
+    /**
+     * The proportion of {@link #poolSize()} connections that are checked approximately
+     * every {@link #timeBetweenConnectionEvictionRunsSeconds()} seconds to see if has been idle at least
+     * {@link #idleConnectionTimeoutSeconds()} seconds and evicts it from the pool if so. For example, given the
+     * the default values, 0.1 * 30 = 3 connections will be checked approximately every 20 seconds and will
+     * be evicted from the pool if it has been idle for at least 10 minutes.
+     */
+    @Value.Default
+    public double proportionConnectionsToCheckPerEvictionRun() {
+        return 0.1;
+    }
+
+    @Value.Default
+    public int idleConnectionTimeoutSeconds() {
```

Comment:
This is the cool-off period during which we keep all of the connections we bursted the pool up to.
He's making it longer (and configurable), iirc because there are some awkward application workflows where the application wakes up on a timer that was slightly longer than this `idleConnectionTimeoutSeconds` before, resulting in it always having to spawn new connections and wait for them.

Cassandra can have closed the connection before this, but it is not especially likely.